deposit-transactions.md

```diff
@@ -0,0 +1,165 @@
+# Deposit Transactions
+
+<!-- External references. -->
+[payload-format]: /rollup-node.md#payload-transaction-format) [EIP-155]:
+<https://eips.ethereum.org/EIPS/eip-155> [EIP-1967]: <https://eips.ethereum.org/EIPS/eip-1967>
+[EIP-2718]: <https://eips.ethereum.org/EIPS/eip-2718> [ABI]:
+<https://docs.soliditylang.org/en/v0.8.10/abi-spec.html>
+
+Although the L2 execution engine is nearly identical to that of L1, an important difference is the
+introduction of a new 'Deposit' Transaction Type. Deposit transactions have the following notable
+distinctions from existing transaction types:
+
+1. They are initiated by the System as part of the protocol.
+2. Do not require signature validation.
+
+## Deposit Transaction Type
+
+We define a new [EIP-2718] compatible transaction type with the prefix `0x7E`, and the following
+fields:
+
+<!-- ToDo: set to more GoLang like type defs? -->
+- `address to`
+- `address from`
+- `uint256 value`
+- `bytes data`
```

Comment:
We agreed to defer on how to handle fees. 
What are the implications of excluding the nonce though? Without a sequencer none I suppose, but worth discussion in the future.

Reply:
Ah, good call, will include that.

Reply:
I know how it works currently. :)  As of now there is no way to make a call on L1 and have a `CREATE` on L2. 
The relevant code path is through the L1 and L2 Cross Domain Messenger contracts, and neither contains a `CREATE` op.
So this is one area in which the new design already makes things better, with much less complexity.

Reply:
Apologies, I forget not everyone here is new 😂 
What (in the current version of Optimistic Ethereum) prevents a deposit from doing a L2 contract call which calls `CREATE`?
Are deposits purely separate from the cross domain messengers? Also for tokens?

Reply:
Historically we have just only implemented cross-contract calls--there is no fundamental reason that creations won't work, it's just about implementing it. Two notes:
- There needs to be a way to indicate intent to create on L1 within the Deposit Feed contract. Options would be a separate function on L1 and a new event field, or just specifying `0x000` as target to do a CREATE.
- We should specify how a deposit transaction is interpreted as a CREATE by the EE, i.e. if `to == nil` (which would be compatible with either of the options in the previous bullet).

I think it's worth specifying this here.

Reply:
FWIW I would advocate for adding a `gasLimit` to this type.  I think it's unlikely that we would not have this field -- even if we defer on how it is populated or validated right now, I think it will exist.  It could just be an argument to the deposit feed's function right now.

Reply:
@ben-chain Could you expand on why we need this for? (Are there any use cases if we don't charge L2 fees on L2 for deposits?)